In [1]:
import dask.dataframe as dd

In [7]:
dfAll = dd.read_csv('*.csv')
dfAll.head()

COMMENT_ID            AUTHOR  \
0  LZQPQhLyRh80UYxNuaDWhIGQYNQ96IuCg-AYWqNPjpU         Julius NM   
1  LZQPQhLyRh_C2cTtd9MvFRJedxydaVW-2sNg5Diuo4A       adam riyati   
2  LZQPQhLyRh9MSZYnf8djyk0gEF9BHDPYrrK-qCczIY8  Evgeny Murashkin   
3          z13jhp0bxqncu512g22wvzkasxmvvzjaz04   ElNino Melendez   
4          z13fwbwp1oujthgqj04chlngpvzmtt3r3dw            GsMega   

                  DATE                                            CONTENT  \
0  2013-11-07T06:20:48  Huh, anyway check out this you[tube] channel: ...   
1  2013-11-07T12:37:15  Hey guys check out my new channel and our firs...   
2  2013-11-08T17:34:21             just for test I have to say murdev.com   
3  2013-11-09T08:28:43   me shaking my sexy ass on my channel enjoy ^_^    
4  2013-11-10T16:05:38            watch?v=vtaRGgvGtWQ   Check this out .   

   CLASS  
0      1  
1      1  
2      1  
3      1  
4      1

In [27]:
dfAll.shape

(Delayed('int-07497720-308e-4222-b4cf-5de63ac4e2b2'), 6)

In [14]:
len(dfAll)

1956

dfAll.head() shows that there are 5 columns and len(dfAll) shows there are 1,956 rows of data in the df.
Below I am persisting the data so I dont lose it in memory.

In [17]:
dfAll = dfAll.persist()

In [19]:
dfAll.CLASS.value_counts().compute()

1    1005
0     951
Name: CLASS, dtype: int64

So we have 1,005 comments labeled as spam('1') and 951 marked as legitimate('0')

Converting CONTENT strings to all lower case letters below

In [20]:
dfAll['CONTENT_lower'] = dfAll.CONTENT.astype(str).str.lower()

In [21]:
dfAll.head()

COMMENT_ID            AUTHOR  \
0  LZQPQhLyRh80UYxNuaDWhIGQYNQ96IuCg-AYWqNPjpU         Julius NM   
1  LZQPQhLyRh_C2cTtd9MvFRJedxydaVW-2sNg5Diuo4A       adam riyati   
2  LZQPQhLyRh9MSZYnf8djyk0gEF9BHDPYrrK-qCczIY8  Evgeny Murashkin   
3          z13jhp0bxqncu512g22wvzkasxmvvzjaz04   ElNino Melendez   
4          z13fwbwp1oujthgqj04chlngpvzmtt3r3dw            GsMega   

                  DATE                                            CONTENT  \
0  2013-11-07T06:20:48  Huh, anyway check out this you[tube] channel: ...   
1  2013-11-07T12:37:15  Hey guys check out my new channel and our firs...   
2  2013-11-08T17:34:21             just for test I have to say murdev.com   
3  2013-11-09T08:28:43   me shaking my sexy ass on my channel enjoy ^_^    
4  2013-11-10T16:05:38            watch?v=vtaRGgvGtWQ   Check this out .   

   CLASS                                      CONTENT_lower  
0      1  huh, anyway check out this you[tube] channel: ...  
1      1  hey guys check out my new channel and our firs...  
2      1             just for test i have to say murdev.com  
3      1   me shaking my sexy ass on my channel enjoy ^_^   
4      1            watch?v=vtarggvgtwq   check this out .

Verifying the count of spam(CONTENT=1) comments that contain 'check' in the message(CONTENT_lower).

In [22]:
verify_spam = dfAll[(dfAll.CLASS == 1) & (dfAll['CONTENT_lower'].str.contains('check'))]
len(verify_spam)

461

In [ ]:
Verifying the count of legitimate(CONTENT=0) comments that contain 'check' in the message(CONTENT_lower).

In [23]:
verify_legit = dfAll[(dfAll.CLASS == 0) & (dfAll['CONTENT_lower'].str.contains('check'))]
len(verify_legit)

19

Part 2. Big Data Options

With the wide variety of technologies that we have been shown and worked with, I still prefer working with Python directly. I think it provides a fairly intuitive and easier to learn introduction to working with data on all scales. It is approachable and yet powerful enough to provide great insights and vizualizations of data. While dask and scala provide similar functionality, I do not feel that they provide the range of features and capabilities as Python does with all of the available libraries that it entails. 

I do feel that online services such as SageMaker(and all of the AWS services, really) and DataBricks are superb in their ability to leverage big computing environments for large scale datasets that could not be handled by a local machine. AWS services offer a wide range that will suit every company's needs, regardless of their growth. Both provide grand computing power but DataBricks seems a little more approachable for the novice and having a free offering makes it viable to get the online benefits at the right price for small businesses and single consumers. SageMaker is really useful though, as it provides a more closely match environment that I am accustom to with Anaconda iPython. 

All in all, having an opportunity to work on all of these technologies has been a great experience and is a good step towards being comfortable in any environment in our future careers. All of these pieces have a place in the data science world and it is nice to be able to talk to the fact that we have exposure to so many useful tools. I hope to revisit most of these items in the near future to expand my own knowledge base.